### PR TITLE
Fix position of Tooltips

### LINKED ISSUE #65947: Tooltip place at wrong place when embed subwindows is enabled
### Godot version

4.0 beta 1

### System information

Fedora x11 NVIDIA GeForce GTX 970

### Issue description

Tooltip_text is placed at a absolute position, (roughly) center of screen when on full screen when 'embed subwindows' is enabled.
Tooltip is not visible when not fullscreen.

Enabled:
Screenshot from 2022-09-16 15-19-10

Enabled full screen:
Screenshot from 2022-09-16 15-18-42

Disabled:
Screenshot from 2022-09-16 15-23-45

### Steps to reproduce

- Create controll node with tool_tip text
- run scene with and without  'embed subwindows' in project settings

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by Sauermann**
resolve #65947

`CanvasItem::get_screen_transform` returns a transform from the CanvasItem to the coordinate system, where a Popup - created as a child of the CanvasItem - should be opened.
`CanvasItem::get_screen_transform` makes some simplifications, that work well, when used in the editor, but not in general cases.

Since Popups like Tooltips are now used more commonly in projects, it becomes necessary to correct these simplifications.

This solution introduces `Viewport::get_popup_base_transform`, which makes the necessary calculations.

The next time, user-facing-function-renames are possible `CanvasItem::get_screen_transform` should be renamed to `CanvasItem::get_popup_transform`.

MRP: [BugPopupTransform.zip](https://github.com/godotengine/godot/files/9998169/BugPopupTransform.zip)

Before and after:

https://user-images.githubusercontent.com/6299227/201543902-a4825700-f9f0-44e1-adb5-6a996ac3499c.mp4

https://user-images.githubusercontent.com/6299227/201543912-f4d0539d-c5c3-402e-a780-17d7e8d6e4d0.mp4

### CODE REVIEW

File: scene/main/viewport.cpp [hunk 1]
```diff
@@ -4183,6 +4183,21 @@ Transform2D SubViewport::get_screen_transform() const {
...
 
+Transform2D SubViewport::get_popup_transform() const {
+	if (is_embedding_subwindows()) {
+		return Transform2D();
+	}
+	SubViewportContainer *c = Object::cast_to<SubViewportContainer>(get_parent());
+	if (!c) {
+		return Viewport::get_screen_transform();
+	}
+	Transform2D container_transform = Transform2D();
```
Comment (akien-mga) [inline scene/main/viewport.cpp:4194]: The assignment is redundant as `Transform2D` is default initialized.

**Comment by akien-mga** [comment]
> The next time, user-facing-function-renames are possible `CanvasItem::get_screen_transform` should be renamed to `CanvasItem::get_popup_transform`.

We can still do renames now if they're important enough. If this helps clean the API and reduce confusion, it's probably best done now than waiting for 5.0.

**Comment by groud** [comment]
Would this PR fix https://github.com/godotengine/godot/issues/71327 ?

**Review by groud — DISMISSED** [review]
Approved in PR meeting. Needs `get_popup_transform` to be renamed `get_base_popup_transform` though.

**Comment by akien-mga** [comment]
> Approved in PR meeting. Needs `get_popup_transform` to be renamed `get_base_popup_transform` though.

Or `get_popup_base_transform`? :D

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**